specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2023-03-01-preview/LongTermRetention.json

```diff
@@ -0,0 +1,411 @@
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "Request body for operation",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/LtrPreBackupRequest"
+            }
+          },
+          {
+            "name": "x-ms-client-request-id",
+            "in": "header",
+            "description": "A unique ID provided by the client to identify the specific request.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Ok. API finished synchronously with Success.",
+            "schema": {
+              "$ref": "#/definitions/LtrPreBackupResponse"
+            },
+            "headers": {
+              "x-ms-request-id": {
+                "description": "A unique ID for the current operation, service generated. All the resource providers must return this value in the response headers to facilitate debugging.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error Scenario 4xx/5xx",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "description": "ErrorCode string in the event of a failure.",
+                "type": "string"
+              },
+              "x-ms-request-id": {
+                "description": "A unique ID for the current operation, service generated. All the resource providers must return this value in the response headers to facilitate debugging.",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Sample_Prebackup": {
+            "$ref": "./examples/LongTermRetentionPreBackup.json"
+          }
+        },
+        "deprecated": false
+      }
+    }
+  },
+  "definitions": {
+    "BackupRequestBase": {
+      "description": "BackupRequestBase is the base for all backup request.",
+      "required": [
+        "backupSettings"
+      ],
+      "type": "object",
+      "properties": {
+        "backupSettings": {
+          "$ref": "#/definitions/BackupSettings",
+          "description": "Backup Settings"
+        }
+      }
+    },
+    "LtrBackupRequest": {
+      "description": "The request that is made for a backup.",
+      "type": "object",
+      "properties": {
+        "targetDetails": {
+          "$ref": "#/definitions/BackupStoreDetails",
+          "description": "Backup store detail for target server"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/BackupRequestBase"
+        }
+      ],
+      "required": [
+        "targetDetails"
+      ]
+    },
+    "LtrPreBackupRequest": {
```

Comment:
>LtrPreBackupRequest [](http://example.com/codeflow?start=5&length=19)

why are targetDetails optional in the validation scenario?

Reply:
targetDetails are not required for validation and they include secrets (SasUri list) which are not required for performing validation. Hence they are not added to request object for validation.

Reply:
>they include secrets (SasUri list) which are not required for performing validation.

They may not be required by your current validations, but it seems likely to me, going out on a limb here, that customers are going to ask you to add more complete validations, so that they can be 100% sure their request is going to succeed, later, and not e.g. have it fail because they gave you the wrong kind of storage account, or their SAS token is expired, or etc. etc.

Reply:
Hey! that definitely seems like a good idea - however this integration pattern was provided by Azure Backup Team, they want a simple API to just check if their server is ready and based on the amount of data the number of storage containers they need to give us to copy the backup. 

The response to pre-backup tells us the number of SAS URIs to provide to copy the backup, which can be provided in the successive backup call. In case the token is expired we will fail the backup fast.